src/helper/selection-tools/scale-tool.js

```diff
@@ -121,6 +121,12 @@ class ScaleTool {
             sx *= signx;
             sy *= signy;
         }
+        if (Math.abs(sx) < 0.0001) {
```

Comment:
Thanks! This seems like a good way to fix the issue!
Rather than retaining lastSx, should sx snap to +-.0001?

Reply:
@fsih I think that would also be possible, the 0.0001 was chosen arbitrary. The difference between the two would normally not be noticeable I think. I can change it if you want. Let me know.

I also wanted to add restricting to whole x/y increment (so only 100%, 200% etc for width cq height) using the ctrl key as a modifier. I know that handy modifier from Corel Draw! in the old days :-) Would you consider a feature like that? Shall I create an issue for that?

Reply:
Hmm, I agree that it's very unlikely to be noticeable, but let's snap to 0.0001 so that behavior is always consistent. Can you also define a constant for .0001, maybe MIN_SCALE_FACTOR?

That feature from Corel Draw sounds handy! Sure, if you create an issue, I can ask our design team what they think.